File: solutions/swift/resistor-color-duo/2/Sources/ResistorColorDuo/ResistorColorDuo.swift
enum ResistorColorDuo: String, CaseIterable {
    case black, brown, red, orange, yellow, green, blue, violet, grey, white
    static func value(for color: String) -> ResistorColorDuo {
        self.init(rawValue: color.lowercased())!
    }
    static func value(for colors: [String]) throws -> Int {
        let (e1, e2) = (value(for: colors[0]), value(for: colors[1]))
        let (v1, v2) = (allCases.firstIndex(of: e1)!, allCases.firstIndex(of: e2)!)
        return v1 * 10 + v2
    }
}

File: solutions/clojure/resistor-color-duo/2/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def color-codes
  {"black" 0 "brown" 1 "red" 2 "orange" 3  "yellow" 4
   "green" 5 "blue" 6 "violet" 7 "grey" 8 "white" 9})

(defn resistor-value [[c1 c2]]
  (+ (* 10 (color-codes c1))
     (color-codes c2)))
